applications/crossbar/src/crossbar_view.erl

```diff
@@ -973,15 +1044,23 @@ filter_foldl(Mapper, [JObj | JObjs], Acc) ->
 %% exhausted yet.
 %% @end
 %%------------------------------------------------------------------------------
--spec last_key(last_key(), kz_json:objects(), non_neg_integer() | 'undefined', non_neg_integer()) ->
+-spec last_key(last_key(), kz_json:objects(), non_neg_integer() | 'undefined', non_neg_integer(), page_size()) ->
                       {last_key(), kz_json:objects()}.
-last_key(LastKey, [], _, _) ->
+last_key(LastKey, [], _Limit, _Returned, _PageSize) ->
+    lager:info("no results same last key ~p", [LastKey]),
     {LastKey, []};
-last_key(LastKey, JObjs, 'undefined', _) ->
-    {LastKey, lists:reverse(JObjs)};
-last_key(LastKey, JObjs, Limit, Returned) when Returned < Limit ->
+last_key(LastKey, JObjs, 'undefined', _Returned, _PageSize) ->
+    lager:info("no limit, re-using last key ~p", [LastKey]),
     {LastKey, lists:reverse(JObjs)};
-last_key(_LastKey, JObjs, Limit, Returned) when Returned == Limit ->
+last_key(_LastKey, JObjs, Limit, Limit, _PageSize) ->
+    lager:info("full page fetched, calculating new key"),
+    new_last_key(JObjs);
+last_key(_LastKey, JObjs, _Limit, _Returned, _PageSize) ->
+    lager:info("returned ~p < limit ~p ps: ~p lk: ~p", [_Returned, _Limit, _PageSize, _LastKey]),
```

Comment:
s/lk/last_key/g is more descriptive.

Reply:
updated texts and downgraded to debug in places